File: Homeworks/C#2/MethodsHomework/06.FirstLargerThanNeighbours/FirstLargerThanNeighbours.cs
/*
Problem 6. First larger than neighbours

Write a method that returns the index of the first element in array that is larger than its neighbours, or -1, if there’s no such element.
Use the method from the previous exercise.
*/


using System;

class FirstLargerThanNeighbours
{
    static int CheckIfLargerThanNeighbours(int[] array)
    {
        int index = -1;
        for (int i = 0; i < array.Length; i++)
        {

            if (i < array.Length - 1 && i > 0)
            {
                if ((array[i] > array[i + 1]) && (array[i] > array[i - 1]))
                {
                    index = i;
                }

            }
        }
        return index; 
    }


    static void Main()
    {
        Console.Write("Please, enter number for size of array: ");
        int size = int.Parse(Console.ReadLine());

        int[] arrayNumbers = new int[size];
        for (int i = 0; i < arrayNumbers.Length; i++)
        {
            Console.Write("Please, enter element[{0}]:", i);
            arrayNumbers[i] = int.Parse(Console.ReadLine());
        }

         int result = CheckIfLargerThanNeighbours(arrayNumbers);
        Console.WriteLine("Index of the first element in array that is larger than its neighbours: {0}", result);
    }
}

File: Homeworks/JavaScriptFundamentals/06.Functions/07.FirstLargerThanNeighbours/FirstLargerThanNeighbours.js
/*
 Write a function that returns the index of the first element in array that is larger than its neighbours or -1, if there’s no such element.
 Use the function from the previous exercise.
 */

var jsConsole,
    arrayNumbers = [5, 12, 8, 33, 40, 21, 58, 44, 89, 23, 76],
    i;

function checkFirstLargerThanNeighbours(array) {
    var index = -1;
    for (i = 0; i < array.length; i += 1) {

        if (i < array.length - 1 && i > 0) {
            if ((array[i] > array[i + 1]) && (array[i] > array[i - 1])) {
                return index = i;
            }
            else {
                return index;
            }

        }
    }
}

jsConsole.writeLine('Test array is [' + arrayNumbers + ']');
console.log('Test array is [' + arrayNumbers + ']');

jsConsole.writeLine('Index of the first element in array that ' +
    'is larger than its neighbours: ' + checkFirstLargerThanNeighbours(arrayNumbers));
console.log('Index of the first element in array that ' +
    'is larger than its neighbours: ' + checkFirstLargerThanNeighbours(arrayNumbers));